src/main/src/main/java/org/geoserver/catalog/CatalogBuilder.java

```diff
@@ -1442,10 +1450,10 @@ WMTSLayerInfo buildWMTSLayer(StoreInfo store, String layerName) throws IOExcepti
         // TODO: handle axis order here ?
         // try to get the native SRS -> we use the bounding boxes, GeoServer will publish all of the
         // supported SRS in the root, if we use getSRS() we'll get them all
-        CoordinateReferenceSystem preferred = layer.getPreferredCRS();
-        if (preferred != null) {
-            wli.setSRS(CRS.toSRS(preferred));
-            wli.setNativeCRS(preferred);
+        if (nativeCRS == null) nativeCRS = layer.getPreferredCRS();
```

Comment:
Confused, since we already have a preferred CRS, why go and use native? Are the two settings redundant?
Where is the native one coming from? (if it's the resource one, it's most of the time there, so the preferred one would be ignored, making it almost useless?)

Reply:
The nativeCRS is picked up to allow computation of native bounds being coherent to the SRS selected by the user: when computing bounds the wmts is rebuilt. Without the information of the nativeCRS the method was pick up a random srs found from the underlying layer here https://github.com/geoserver/geoserver/blob/624645ccaec2ac32e28742a3740f073ea144bf76/src/main/src/main/java/org/geoserver/catalog/CatalogBuilder.java#L1459 and retrieving bbox down using that one https://github.com/geoserver/geoserver/blob/624645ccaec2ac32e28742a3740f073ea144bf76/src/main/src/main/java/org/geoserver/catalog/CatalogBuilder.java#L1498.
The preferred CRS is a concept coming from geotools and is unclear to what it should be used for, was a leftover in that CatalogBuilder code my fault, I've removed it.